packages/api/lib/pdrHelpers.js

```diff
@@ -17,6 +18,61 @@ function generateShortPAN(disposition) {
   );
 }
 
+/**
+ * Get list of input granules from execution
+ *
+ * @param {string} executionArn - execution arn
+ * @returns {object[]} list of granules
+ */
+async function getGranulesFromExecution(executionArn) {
+  const excObj = await getExecution({
+    prefix: process.env.stackName,
+    arn: executionArn,
+  });
+  return excObj.originalPayload.granules;
+}
+
+const granulesFileCount = (granules) => granules.reduce((sum, { files }) => sum + files.length, 0);
+
+/**
+ * Generate Long PAN message
+ *
+ * @param {object|string[]} executions - List of workflow executions
+ * @returns {string} the PAN message
+ */
+async function generateLongPAN(executions) {
+  const timeStamp = new Date();
+
+  const executionsWithGranules = [];
+  /* eslint-disable no-await-in-loop */
+  for (const exc of executions) {
```

Comment:
Just on the other side of a nit: 

This probably shouldn't be a for loop, we can do this asynchronously for speed reasons given the granulesFromExecutions call. 

e.g. 

  await Promise.all(executions.map((exec) => {
    const arn = exec?.arn || exec // at this point in review, not sure why arn || exc is here
    const granules = await getGranulesFromExecution(arn)
    executionsWithGranules.push({ arn, granules, reason: exc.reason });

  }));

  For larger granule counts it's probably not an insignificant wait time.